File: pa3/FilePaths.java
public final class FilePaths {

	public static final String DATA_DIRECTORY = "Data/";
	public static final String OUTPUT_DIRECTORY = DATA_DIRECTORY + "OutputJava/";
	public static final String HUFFMAN_IO_TEST = DATA_DIRECTORY + "testHuffmanIO";
	public static final String BWT_COMPRESSED_DIRECTORY = OUTPUT_DIRECTORY + "BWT/";
	public static final String LZ78_COMPRESSED_DIRECTORY = OUTPUT_DIRECTORY + "LZ78/";
	public static final String LZ77_COMPRESSED_DIRECTORY = OUTPUT_DIRECTORY + "LZ77/";
	public static final String LZSS_COMPRESSED_DIRECTORY = OUTPUT_DIRECTORY + "LZSS/";

	public static final String DATA[] = {
			"tiny.txt",
			"medium.txt",
			"aliceinwonderland.txt",
			"H.pylori.fna",
			"L.monocytogenes.fna",
			"bible.txt",
			"E.Coli.fna",
			"S.cerevisiae.fna",
			"N.crassa.fna",
			"C.elegans.fna" };
	public static final int NUM_DATA = 10;

	/**
	 * Large DNA file. You do not have to run this.
	 */
//	 public static final String DATA[] = { "T.nigroviridis.fna"};
//	 public static final int NUM_DATA = 1;

	public static final String ALPHABET_EXTENSION = ".alpha";

	public static final String HUFFMAN_ENCODING_EXTENSION = ".huffman.encoding";
	public static final String HUFFMAN_MAP_EXTENSION = ".huffman.map";

	public static final String LZ78_NODE_ENCODING_EXTENSION = ".nodes";
	public static final String LZ78_CHARS_EXTENSION = ".chars";

	public static final String LZ77_DELTA_EXTENSION = ".delta";
	public static final String LZ77_LENGTH_EXTENSION = ".length";
	public static final String LZ77_NEXT_EXTENSION = ".next";

	public static final String LZSS_IDENTIFIER_EXTENSION = ".identifier";
	public static final String LZSS_DELTA_EXTENSION = ".delta";
	public static final String LZSS_LENGTH_EXTENSION = ".length";
	public static final String LZSS_NEXT_EXTENSION = ".next";
}

File: pa3/FilePaths.cpp
#include "FilePaths.h"

const string FilePaths::DATA_DIRECTORY = "Data/";
const string FilePaths::OUTPUT_DIRECTORY = DATA_DIRECTORY + "OutputC++/";
const string FilePaths::HUFFMAN_IO_TEST = FilePaths::DATA_DIRECTORY + "testHuffmanIO";

const string FilePaths::BWT_COMPRESSED_DIRECTORY = OUTPUT_DIRECTORY + "BWT/";
const string FilePaths::LZ78_COMPRESSED_DIRECTORY = OUTPUT_DIRECTORY + "LZ78/";
const string FilePaths::LZ77_COMPRESSED_DIRECTORY = OUTPUT_DIRECTORY + "LZ77/";
const string FilePaths::LZSS_COMPRESSED_DIRECTORY = OUTPUT_DIRECTORY + "LZSS/";

const string FilePaths::DATA[] = {
    "tiny.txt",
    "medium.txt",
    "aliceinwonderland.txt",
    "H.pylori.fna",
    "L.monocytogenes.fna",
    "bible.txt",
    "E.Coli.fna",
    "S.cerevisiae.fna",
    "N.crassa.fna",
    "C.elegans.fna" };

const int FilePaths::NUM_DATA = 10;

/**
 * Large DNA file. You do not have to run this.
 */
//const string FilePaths::DATA[] = { "T.nigroviridis.fna" };
//const int FilePaths::NUM_DATA = 1;


const string FilePaths::ALPHABET_EXTENSION = ".alpha";

const string FilePaths::HUFFMAN_ENCODING_EXTENSION = ".huffman.encoding";
const string FilePaths::HUFFMAN_MAP_EXTENSION = ".huffman.map";

const string FilePaths::LZ78_NODE_ENCODING_EXTENSION = ".nodes";
const string FilePaths::LZ78_CHARS_EXTENSION = ".chars";

const string FilePaths::LZ77_DELTA_EXTENSION = ".delta";
const string FilePaths::LZ77_LENGTH_EXTENSION = ".length";
const string FilePaths::LZ77_NEXT_EXTENSION = ".next";

const string FilePaths::LZSS_IDENTIFIER_EXTENSION = ".identifier";
const string FilePaths::LZSS_DELTA_EXTENSION = ".delta";
const string FilePaths::LZSS_LENGTH_EXTENSION = ".length";
const string FilePaths::LZSS_NEXT_EXTENSION = ".next";
